monkey/monkey_island/cc/services/representations.py

```diff
@@ -1,47 +1,33 @@
 from datetime import datetime
 from enum import Enum
+from json import loads
 
 import bson
-from bson.json_util import dumps
 from flask import make_response
+from flask.json import JSONEncoder, dumps
 
+from common.utils import IJSONSerializable
 
-def _normalize_obj(obj):
-    if ("_id" in obj) and ("id" not in obj):
-        obj["id"] = obj["_id"]
-        del obj["_id"]
 
-    for key, value in list(obj.items()):
-        obj[key] = _normalize_value(value)
-    return obj
-
-
-def _normalize_value(value):
-    # ObjectId is serializible by default, but returns a dict
-    # So serialize it first into a plain string
-    if isinstance(value, bson.objectid.ObjectId):
-        return str(value)
-
-    if isinstance(value, list):
-        return [_normalize_value(_value) for _value in value]
-    if isinstance(value, tuple):
-        return tuple((_normalize_value(_value) for _value in value))
-    if type(value) == dict:
-        return _normalize_obj(value)
-    if isinstance(value, datetime):
-        return str(value)
-    if issubclass(type(value), Enum):
-        return value.name
-
-    try:
-        dumps(value)
-        return value
-    except TypeError:
-        return value.__dict__
+class APIEncoder(JSONEncoder):
+    def default(self, value):
+        # ObjectId is serializible by default, but returns a dict
+        # So serialize it first into a plain string
+        if isinstance(value, bson.objectid.ObjectId):
+            return str(value)
+        if isinstance(value, datetime):
+            return str(value)
+        if issubclass(type(value), Enum):
+            return value.name
+        if issubclass(type(value), IJSONSerializable):
+            return loads(value.__class__.to_json(value))
+        try:
+            return JSONEncoder.default(self, value)
+        except TypeError:
+            return value.__dict__
```

Comment:
If we return `value.__dict__`, will each item in the dictionary be encoded?

Reply:
Yes